WordPress/Classes/ApplicationToken/ApplicationTokenListView.swift

```diff
@@ -0,0 +1,149 @@
+import SwiftUI
+import DesignSystem
+import WordPressUI
+import WordPressAPI
+
+struct ApplicationTokenListView: View {
+
+    @ObservedObject
+    private var viewModel: ApplicationTokenListViewModel
+
+    fileprivate init(tokens: [ApplicationTokenItem]) {
+        let dataProvider = StaticTokenProvider(tokens: .success(tokens))
+        self.init(viewModel: ApplicationTokenListViewModel(dataProvider: dataProvider))
+    }
+
+    fileprivate init(error: Error) {
+        let dataProvider = StaticTokenProvider(tokens: .failure(error))
+        self.init(viewModel: ApplicationTokenListViewModel(dataProvider: dataProvider))
+    }
+
+    init(viewModel: ApplicationTokenListViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        VStack {
+            if viewModel.isLoadingData {
+                ProgressView()
+            } else if let error = viewModel.errorMessage {
+                EmptyStateView(Self.errorTitle, systemImage: "exclamationmark.triangle", description: error)
+            } else {
+                List(viewModel.applicationTokens) { token in
+                    ApplicationTokenListItemView(item: token)
+                }
+                .listStyle(.plain)
+            }
+        }
+        .navigationTitle(Self.title)
+        .navigationBarTitleDisplayMode(.inline)
+        .onAppear {
+            if viewModel.applicationTokens.isEmpty {
+                Task {
+                    await viewModel.fetchTokens()
+                }
+            }
+        }
+    }
+}
+
+class ApplicationTokenListViewModel: ObservableObject {
+
+    @Published
+    var isLoadingData: Bool = false
```

Comment:
(nit) applies to the remaining properties

```suggestion
    private(set) var isLoadingData: Bool = false
```